SYCL/ESIMD/regression/atomic_update_test.cpp

```diff
@@ -0,0 +1,46 @@
+//==---------------- atomic_smoke.cpp  - DPC++ ESIMD on-device test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This test checks regression in lsc_atomic_update
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include "../esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+
+int main() {
+  sycl::queue q{};
+  auto p_sync = malloc_shared<uint32_t>(1024, q);
+  p_sync[0] = 5;
+  p_sync[1] = 0;
+  sycl::nd_range<1> range(sycl::range<1>(256 * 4), sycl::range<1>(256));
+  q.submit([&](sycl::handler &cgh) {
+    cgh.parallel_for<class reproducer>(
+        range, [=](sycl::nd_item<1> idx) [[intel::sycl_explicit_simd]] {
+          uint32_t result =
+              sycl::ext::intel::experimental::esimd::lsc_atomic_update<
+                  sycl::ext::intel::esimd::atomic_op::load, uint32_t, 1>(
+                  p_sync, 0, 1)[0];
+          p_sync[1] = result;

```

Comment:
- Writing with many thread to 1 address/variable is UB in general.
  If you want to have only 1 thread, please use cgh.single_task()

- Formally speaking, in ESIMD mode the memory must be written via simd objects or atomic_update.